modules/openapi-generator/src/main/resources/typescript-axios/common.mustache

```diff
@@ -80,7 +80,14 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
     const searchParams = new URLSearchParams(url.search);
     for (const object of objects) {
         for (const key in object) {
-            searchParams.set(key, object[key]);
+            if (Array.isArray(object[key])) {
+                searchParams.delete(key);
```

Comment:
What's the reason this was added? 

If this is to avoid duplicate keys for different objects, we should IMO instead raise an error instead of silently removing the previously set object.

Reply:
It's to make the array behaviour behave the same as the non-array behaviour.

`set` replaces any current query arguments. So the current non-array behaviour of setSearchParams is to delete any existing query argument when it replaces it. `append` doesn't clear previous values so I included a `delete` to ensure that `{foo: 'bar'}` and `{foo: ['bar']}` do not exhibit different behaviours, as that would be unexpected.

Reply:
Agreed on the behavior of `set` vs `append`. Why do we need `searchParams.delete(key)` before the `for` loop that uses `append`?

The case I have in mind is the following:
If `objects = [{foo: 'bar'}, {foo: ['baz', 'bazzzz']}]`, then the code currently would 
* `set("foo", "bar")`
* `delete("foo")`
* append("foo", "baz")
* append("foo", "bazzz")

Shouldn't we retain the `foo` from `objects[0]` or preferably throw an error because this is entirely undefined behavior?

Obviously, overlapping keys shouldn't happen but I'd prefer throwing an error in that case instead of silently dropping the first key.

Reply:
In the current implementation that object will do the following:

- `set("foo", "bar")`
- `set("foo", ['baz', 'bazzzz'])`

Which will result in a query string of `foo=baz,bazzzz`. The current behaviour is that any key overwrites any previous value.

The comma-ness is a critical bug, it results in `style: form, explode: true` in the OpenAPI schema result in `foo=baz,bazzzz` when according to the OpenAPI spec it's supposed to result in ``foo=baz&foo=bazzzz`.

I am trying to fix this bug, ensuring that an array correctly results in setting multiple `key=` query parameters instead of incorrectly stringifying the array.

I'm not completely opposed to making `setSearchParams` always append (however in that case the `set` for non-arrays should also be `append`). However, that would technically be a breaking change (AFAICT making query options append instead of overwrite OpenAPI parameters).

I am trying to fix a critically broken bug in the typescript-axios generator that makes it unusable if you have a `style: form, explode: true` OpenAPI spec. So I do not want to introduce any breaking changes that would delay the release of the bugfix.

Reply:
:man_facepalming: Thanks for clearing this up! :) Let's retain the old behavior, for now, to not break any current setups.